core/templates/dev/head/services/SearchService.ts

```diff
@@ -16,206 +16,225 @@
+  }
 
-      var selections = selectionDetails[itemsType].selections;
-      for (var i = 0; i < items.length; i++) {
-        selections[items[i]] = true;
-      }
-    };
-
-    var _isCurrentlyFetchingResults = false;
-    var numSearchesInProgress = 0;
-
-    var getQueryUrl = function(searchUrlQueryString) {
-      return SEARCH_DATA_URL + '?q=' + searchUrlQueryString;
-    };
-
-    return {
-      getSearchUrlQueryString: function(searchQuery, selectedCategories,
-          selectedLanguageCodes) {
-        return encodeURIComponent(searchQuery) +
-          _getSuffixForQuery(selectedCategories, selectedLanguageCodes);
-      },
-      // Note that an empty query results in all activities being shown.
-      executeSearchQuery: function(
-          searchQuery, selectedCategories, selectedLanguageCodes,
-          successCallback) {
-        var queryUrl = getQueryUrl(
-          this.getSearchUrlQueryString(
-            searchQuery, selectedCategories, selectedLanguageCodes));
-
-        _isCurrentlyFetchingResults = true;
-        numSearchesInProgress++;
-        $http.get(queryUrl).then(function(response) {
-          var data = response.data;
-          _lastQuery = searchQuery;
-          _lastSelectedCategories = angular.copy(selectedCategories);
-          _lastSelectedLanguageCodes = angular.copy(selectedLanguageCodes);
-          _searchCursor = data.search_cursor;
-          numSearchesInProgress--;
-
-          $rootScope.$broadcast(
-            'initialSearchResultsLoaded', data.activity_list);
-          _isCurrentlyFetchingResults = false;
-          var checkMismatch = function(searchQuery) {
-            var isMismatch = true;
-            $('.oppia-search-bar-input').each(function(index) {
-              if ((<string>$(this).val()).trim() === searchQuery) {
-                isMismatch = false;
-              }
-            });
-            return isMismatch;
-          };
-          if (checkMismatch(searchQuery)) {
-            $log.error('Mismatch');
-            $log.error('SearchQuery: ' + searchQuery);
-            $log.error('Input: ' + (
-              <string><any>$('.oppia-search-bar-input').val()).trim());
+  getSearchUrlQueryString(searchQuery: string, selectedCategories: Object,
+      selectedLanguageCodes: Object): string {
+    return encodeURIComponent(searchQuery) +
+        this._getSuffixForQuery(selectedCategories, selectedLanguageCodes);
+  }
+  // Note that an empty query results in all activities being shown.
+  executeSearchQuery(
+      searchQuery: string, selectedCategories: Object,
+      selectedLanguageCodes: Object,
+      successCallback: Function | undefined): boolean | void {
+    let queryUrl = this.getQueryUrl(
+      this.getSearchUrlQueryString(
+        searchQuery, selectedCategories, selectedLanguageCodes));
+
+    this._isCurrentlyFetchingResults = true;
+    this.numSearchesInProgress++;
+    this.httpClient.get(queryUrl).subscribe((response: any) => {
```

Comment:
What is the type for `response`?

Reply:
I am not sure right now, it's data type will be a complex object from the backend, so for now i'll add a Todo 7165

Reply:
that's fine.